From 79c5ac8db1f2685c9a4167e53e3b4794034b4dc8 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Tue, 25 May 2021 16:04:00 +0530
Subject: [PATCH] Apply the correct stride for HEVC codec

Remember the stride decoded from the header, so that
we can use the correct stride information when decoding
the frame. Currently since we donot remember this, there
is a mismatch with the stride we know and stride applied
during decode frame.

Tracked-On: OAM-95490
Signed-off-by: Vinay Kompella <vinay.kompella@intel.com>
Change-Id: I08ab2064d9d311ab0f5a3899a294ae9725af97d9
---
 media/codec2/components/hevc/C2SoftHevcDec.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/media/codec2/components/hevc/C2SoftHevcDec.cpp b/media/codec2/components/hevc/C2SoftHevcDec.cpp
index 6bcf3a24d4..bc6dc8e167 100644
--- a/media/codec2/components/hevc/C2SoftHevcDec.cpp
+++ b/media/codec2/components/hevc/C2SoftHevcDec.cpp
@@ -917,7 +917,8 @@ void C2SoftHevcDec::process(
         if (0 < ps_decode_op->u4_pic_wd && 0 < ps_decode_op->u4_pic_ht) {
             if (mHeaderDecoded == false) {
                 mHeaderDecoded = true;
-                setParams(ALIGN32(ps_decode_op->u4_pic_wd), IVD_DECODE_FRAME);
+                mStride = ALIGN32(ps_decode_op->u4_pic_wd);
+                setParams(mStride, IVD_DECODE_FRAME);
             }
             if (ps_decode_op->u4_pic_wd != mWidth ||  ps_decode_op->u4_pic_ht != mHeight) {
                 mWidth = ps_decode_op->u4_pic_wd;
-- 
2.17.1

